File: strategies/suicidal_cpp/strategies.cpp
#include <iostream>
using namespace std;

int main() {
	char line[81];
	bool isAnt = false;
	while (cin.good()) {
		cin.getline(line, sizeof(line));
		if (line == string("END")) {
			if (isAnt) {
				cout << "SUICIDE\n";
				isAnt = false;
			}
			cout << "END\n";
			cout.flush();
		} else if (line == string("BEGIN ANT")) {
			isAnt = true;
		}
	}
}

File: strategies/suicidal_java/Strategies.java
import java.util.Scanner;

class Strategies {
	public static void main(String[] args) {
		Scanner s = null;
		Boolean isAnt = false;
		try {
			s = new Scanner(System.in);
			while (s.hasNextLine()) {
				String line = s.nextLine();
				if (line.equals("END")) {
					if (isAnt) {
						System.out.println("SUICIDE");
						isAnt = false;
					}
					System.out.println("END");
					System.out.flush();
				} else if (line.equals("BEGIN ANT")) {
					isAnt = true;
				}
			}
		} finally {
			if (s != null) {
				s.close();
			}
		}
	}
}
